datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/DecimalUtils.java

```diff
@@ -92,7 +94,7 @@ public static BigDecimal toBigDecimal(long seconds, int nanoseconds)
             }
             return BigDecimal.valueOf(seconds).setScale(9);
         }
-        return new BigDecimal(toDecimal(seconds, nanoseconds));
+        return BigDecimalParser.parse(toDecimal(seconds, nanoseconds));
```

Comment:
Ok, one suggestion here (which I probably should have made earlier) -- I think this should call

    NumberInput.parseBigDecimal(....)

instead of directly referencing `BigDecimalParser`.